### PR TITLE
[Command] #gearup Table Auto-Install

### PR DESCRIPTION
**Pull request by Akkadius**
**Description**

When using the #gearup command; it would originally instruct you to install the table yourself if it did not exist. Now the command will automatically reach out to Github and install the table for you via HTTPS client request

Gearup is a command to easily add era-contextual equipment to a character. See original commit if interested: https://github.com/EQEmu/Server/pull/1094/commits/cfa3e88bb29f164751e147eca9a57ff128c5316a

Referenced SQL https://raw.githubusercontent.com/EQEmu/Server/master/utils/sql/git/optional/2020_07_20_tool_gearup_armor_sets.sql

image

**Demo Video**

https://www.youtube.com/watch?v=baPBZhNiGZ8

### CODE REVIEW

**Comment by KinglyKrab** [comment]
Seems to have caught the other PR's code, too.

**Comment by dencelle** [comment]
couldn't we use https://github.com/EQEmu/Server/blob/22333ee40b41d5f66ea81bbcca92574341e8d90c/zone/client_packet.cpp#L14060 for this? would reduce how many # commands we have and it would actually give functionality to this command

**Comment by Akkadius** [comment]
> Seems to have caught the other PR's code, too.

Answered in Discord, intentional as relies on httplib

**Comment by Akkadius** [comment]
> couldn't we use
> 
> https://github.com/EQEmu/Server/blob/22333ee40b41d5f66ea81bbcca92574341e8d90c/zone/client_packet.cpp#L14060
> 
> for this? would reduce how many # commands we have and it would actually give functionality to this command

We could, however #gearup is a bit more interactive and testbuff would include more than just gearing up a player

File: world/world_server_command_handler.cpp [hunk 1]
```diff
@@ -291,6 +295,30 @@ namespace WorldserverCommandHandler {
 			return;
 		}
 
+		httplib::Client cli("https://raw.githubusercontent.com");
+		cli.set_connection_timeout(0, 15000000); // 15 sec
+		cli.set_read_timeout(15, 0); // 15 seconds
+		cli.set_write_timeout(15, 0); // 15 seconds
+
+		std::string url = "/EQEmu/Server/master/utils/sql/git/optional/2020_07_20_tool_gearup_armor_sets.sql";
+		if (auto    res = cli.Get(url.c_str())) {
```
Comment (KinglyKrab) [inline world/world_server_command_handler.cpp:304]: Formatting.

File: world/world_server_command_handler.cpp [hunk 2]
```diff
@@ -469,7 +497,7 @@ namespace WorldserverCommandHandler {
 			"destination_character_name",
 			"destination_account_name"
 		};
-		std::vector<std::string> options   = { };
+		std::vector<std::string> options   = {};
```
Comment (KinglyKrab) [inline world/world_server_command_handler.cpp:500]: Formatting.

File: zone/command.cpp [hunk 3]
```diff
@@ -3104,13 +3107,57 @@ void command_gearup(Client *c, const Seperator *sep)
...
+
+		// http get request
+		httplib::Client cli("https://raw.githubusercontent.com");
+		cli.set_connection_timeout(0, 15000000); // 15 sec
+		cli.set_read_timeout(15, 0); // 15 seconds
+		cli.set_write_timeout(15, 0); // 15 seconds
+
+		int sourced_queries = 0;
+		std::string url = "/EQEmu/Server/master/utils/sql/git/optional/2020_07_20_tool_gearup_armor_sets.sql";
+		if (auto    res = cli.Get(url.c_str())) {
```
Comment (KinglyKrab) [inline zone/command.cpp:3125]: Formatting.

File: zone/command.cpp [hunk 4]
```diff
@@ -3158,17 +3208,35 @@ void command_gearup(Client *c, const Seperator *sep)
 		}
 
 		if (equipped.find(slot_id) == equipped.end()) {
-			if (c->CastToMob()->CanClassEquipItem(item_id)) {
+			const EQ::ItemData* item = database.GetItem(item_id);
+
+			bool can_wear_item = !c->CheckLoreConflict(item) &&
```
Comment (KinglyKrab) [inline zone/command.cpp:3213]: Formatting.

File: world/world_server_command_handler.cpp (hunk 1 above)
Comment (Akkadius) [reply world/world_server_command_handler.cpp:304]: This is just variable alignment

File: world/world_server_command_handler.cpp (hunk 2 above)
Comment (Akkadius) [reply world/world_server_command_handler.cpp:500]: This is normal

File: zone/command.cpp (hunk 3 above)
Comment (Akkadius) [reply zone/command.cpp]: This is just variable alignment

File: zone/command.cpp (hunk 4 above)
Comment (Akkadius) [reply zone/command.cpp]: Rewrote this one to read a bit better

**Review by KinglyKrab — APPROVED** [review]

**Review by splose — APPROVED** [review]
Nice

**Pull request merged**